enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1009 @@
+
+  The default IngressController cannot use the EIPs from the cluster config if a custom
+  IngressController is using them. The expectation is that when the operator creates (or recreates) 
+  the default IngressController, it uses any EIPs that are specified in the cluster config.
+  Therefore, a custom IngressController should avoid using the EIPs from the cluster config
+  so as to avoid preventing the default IngressController from using them.
+
+  If the cluster-admin creates 2 custom IngressControllers, then only one of them could use the EIPs
+  from the cluster config, even if the default IngressController weren't using them.  Using the cluster
+  config in this case would mean the outcome would depend on which custom IngressController were created
+  first. Avoiding using the cluster config for custom IngressControllers makes the behavior more predictable.
+
+9. Are the eips mentioned in the cluster config default for the default `IngressController` ?  
+   What happens if the cluster-admin deletes the default IngressController and then recreates it before 
+   the polling loop of the operator does, and the cluster-admin doesn't specify eipAllocations?
+
+  No. The eips in the cluster config are only used when the operator creates the default IngressController.
+  If the default `IngressController` is not found and if the polling loop kicks in and detects it's absence it will
+  create a default IC with the eips mentioned in the cluster config object of ingress.
+  However, it isn't used as a default value when eipAllocations is not specified on the IngressController because
+  the cluster-admin may delete the default IngressController and then recreate it with no eips or no already assigned eips
+  or before the polling loop does.
+  Because the operator only uses the cluster config to specify eipAllocations when the operator itself creates
+  the default IngressController, the cluster-admin can always delete and recreate the default IngressController
+  with eipAllocations deliberately left unspecified, so that the IngressController doesn't use the EIPs from the
+  cluster config. Or the cluster-admin  can always delete and recreate the default IngressController with eipAllocations
+  which are not already specified in the config ingress cluster object.
+
+10. Can you assign `eipAllocations` to an `IngressController` with scope `Internal` ?
+    No, eipAllocations can be provided only for an `IngressController` with scope `External`.
+    The default IC is set to [External](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/operator.go#L476)
+    However, we will validate the `IngressController` CR at API using CEL to check if `eipAllocations` are provided only when `scope` is 
+    set to `External`.
+
+## Test Plan
+
+### This EP will be covered by unit tests in the following:
+- cluster-ingress-operator
+- installer
+
+### This  EP will also be covered by end to end tests:
```

Comment:
Are we planning to add an e2e to installer? It would be nice at least to exercise the managed VPC scenario.

The job [`e2e-aws-ovn-public-ipv4-pool`][job] and step [`ipi-conf-aws-byo-ipv4-pool-public`][step] were created to similar purpose, if you would like to explore it.

[job]: https://github.com/openshift/release/blob/master/ci-operator/config/openshift/installer/openshift-installer-release-4.16__altinfra.yaml#L155-L171
[step]: https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/aws/byo-ipv4-pool-public/ipi-conf-aws-byo-ipv4-pool-public-ref.yaml